WildMeOrg/codex-frontend
# Dex 1066 recoverable user manager deleted collab

# QA Notes

Mutually-revoked collaborations are no longer hidden from the collaboration table.

If either of the collaborations (user1 → user2 or user2 → user1) is revoked, this ticket makes the table display a restore icon instead of a delete icon.

Upon click, the collaboration is restored.

<img width="1552" alt="Screen Shot 2022-05-30 at 10 12 20 AM" src="https://user-images.githubusercontent.com/2775448/171038150-c60fe714-91e3-484e-b3ef-726d9b853bc2.png">
<img width="1552" alt="Screen Shot 2022-05-30 at 10 12 28 AM" src="https://user-images.githubusercontent.com/2775448/171038167-45075bb2-700c-46f3-ac44-ce8f6549f9f7.png">


# Notes for code review

Note that useEstablishCollaborationAsUserManager does not work on collaborations that are in a revoked state already, so usePatchCollaboration was required instead. 

In fact, in order to restore the collaboration, two separate calls to usePatchCollaboration were required (one for each of the two collaborators), thereby necessitating: 
1. usage of Promise.all to handle the success logic and 
2. removal of onClearPatchSuccess and patchSuccess (because we are unable to distinguish between usePatchCollaboration calls for revocation and for restoration. 

Because of 2), I decided to handle the success logic with two useStates; it's a departure from our pattern of using the error and success handling built into usePost/useMutate, but I think that the departure is warranted given the uniqueness of needing usePatchCollaboration to do different things on this same component.
